app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala

```diff
@@ -54,6 +54,18 @@ case class WalletRoutes(wallet: AnyHDWalletApi)(implicit
         }
       }
 
+    case ServerCommand("walletinfo", _) =>
```

Comment:
so to nit pick a bit. The `rootXPub` is associated with the key manager. Since [wallet does depend on key manager](https://github.com/bitcoin-s/bitcoin-s/blob/b0b56dd5da2f7e3c2428008d9a3c9f7391cf53e1/build.sbt#L538) the structure should look something like this.

```
{
  "wallet" : { 
      "keyManager" : { 
           "rootXPub" : xpub
       }
       "walletXpub" : wallet_xpub
       "hdPath" : hd_path 
   }
}
```

and then we have define `keymanagerinfo` which just returns the json object defined by the `keyManager` key. 

That way we get nice composability of our `info`s if we add them for other modules to -- and the configurations represent the dependency graph for the bitcoin-s proejct (`wallet.dependsOn(keyManager)` in this case). 

If we were to add a new field to `keymanagerinfo` later on -- like `isHardwareWallet : true`, it would magically added to the `wallet` configuration as well so our `walletinfo` would look like! 

```
{
  "wallet" : { 
      "keyManager" : { 
           "rootXPub" : xpub
            "isHardwareWallet" : true
       }
       "walletXpub" : wallet_xpub
       "hdPath" : hd_path 
   }
}
```

Which is really nice IMO and we don't have to  change `walletinfo`

Reply:
This seems a bit redundant, technically all this info is about the key manager. I think to the user it makes more sense for them to not need to care about the distiction from the key manager and the wallet.

Reply:
`hdPath` and `wallet_xpub` are not associated or known by the key manager.

If you look at the code, you see 

```scala
wallet.keyManager.getRootXPub
accountDb.xpub
accountDb.hdAccount
```

the later two pieces of information are stored by the wallet.